### DISCUSSION TITLE
Unset GIT_COMMITTER_NAME and GIT_COMMITTER_EMAIL in Codespaces workspace (2022-06)

### DISCUSSION THREAD

**Discussion by tnir** [community/community#20925] [Codespaces] [answered]
> Original post: [Jun 18, 1:33 PM](https://github.community/t/unset-git-committer-name-and-git-committer-email-in-codespaces-workspace/257046)

I would like GitHub to unset `GIT_COMMITTER_NAME` and `GIT_COMMITTER_EMAIL` in a Codespaces’ workspace.

I always need to unset these environment variables in a new Codespaces’ workspace.

```
$ env | grep ^GIT_C
GIT_COMMITTER_NAME=GitHub
GIT_COMMITTER_EMAIL=noreply@github.com
```
If you try to unset them by yourself, you will run into a problem as described below:

cf. [Failed to create a Codespace](https://github.com/github-community/community/discussions/20924)

**Comment by tnir** [comment, marked as answer]
Today I found the problem is resolved now... Anyone knows?

```
$ env | grep ^GIT_C | wc -l
0
```

**Reply by douglascamata** [reply]
I know that unsetting these env vars fixes the issue I reported in #20599, so possibly they fixed my problem.

**Reply by tnir** [reply]
Yeah, I have reported 10 months ago as #20924 (or originally posted at https://github.community/t/failed-to-create-a-codespace/200326), but it looks like GitHub team fixed it very recently!

**Reply by douglascamata** [reply]
I rebuilt my container in an already existing Codespace and the env vars are still there. 

Also created a new Codespace and they were there still.

**Reply by douglascamata** [reply]
@tnir I don't think this is globally solved yet. Can you try out creating a new Codespace in a random repo that doesn't have devcontainer steps to unset those vars? i.e. I just tested in https://github.com/parca-dev/parca-agent and the env vars were present.

**Reply by tnir** [reply]
@douglascamata I tested the repo you showed above, and it works **to me**. GitHub might be rolling out this improvement by user. I hope you get this bugfix soon! (As 2 day passed, you might already got it)

**Reply by bam80** [reply]
I still see these vars in my workspace, has the problem returned?
https://github.com/orgs/community/discussions/20599#discussioncomment-10783057

**Reply by bam80** [reply]
> devcontainer steps to unset those vars?

@douglascamata could you elaborate which are those?
